### DISCUSSION TITLE
Some interaction issues

### DISCUSSION THREAD

**Discussion by quank123wip** [Milkdown/milkdown#354] [Ideas]
I'm not sure if these issues comes from my misunderstand or it is actually a issue so I post them in discussions for a friendly attitude. 🧐 Sorry for troubling anyone for my fault.

1. Formula Blocks Escaping

If I create a formula block, and I finished editing this one, I can escape by Mod-Enter or mouse and jump to the next line. But currently I can't find out a way with keyboard only to move my cursor above the formula block. I wonder if I missed something or it's a issue in interaction indeed.

2. Inline Code input

There is a keymap Mod-E for inline code, which is neat for advanced users. But for people who has no or just Typora or Marktext experiences, they prefer to enter ``` and close it like inserting code blocks in milkdown to insert a inline code. Is there a way to implentment this feature, or it's already exists and I missed it again?

3. Delete a form(or other blocks)

When I was using Marktext, I can delete a empty form by just a backspace, but it seems complicated in milkdown. Maybe it's another feature already in the doc. Currently I'm not familiar with Prosemirror, I wonder if we can delete a form or a formula blocks (seems like marmaid too) like other editors do? Also, is it a better interaction design? Should we make it into milkdown?

4. Remove a heading style

I have made a PR about it, there is currently 2 ways to do this in default config. But it can be improved more than that. Typora and Marktext provided a good way to do this: navigate to the front of this line, the editor will show you a "#" icon to mention you this is a heading block, backspace, and we are done, it's plain text now. It's an intuitive interactive design. I wonder if we can do this in milkdown?

🙏Hope this is not a waste of your precious time, and sorry for discuss it before make a PR because I'm new to projects like this. If they are indeed a improve for this project, I would make a PR if I can(hopefully).✨

**Comment by Saul-Mirone** [comment]
#1 and #3 are good points. I think we can investigate how to improve this behavior. They're all related to the inner editor of milkdown. I plan to extract them into a common component.

#2 You can type ` + code + ` to create a inline code. The limitation here is that you cannot type two ` first because prosemirror's inputrule doesn't support this behavior.

#4 I think it's something nice to have, basically it's a custom view for heading node. Will give it a try when I have time. It's also related to something I want to do before. Which is that create some custom views for nodes to make milkdown can behave like Typora.

**Reply by quank123wip** [reply]
About #4, should we implentment a inner editing feature like showing user identifiers such as ```###```? It should solve #364 case A. But it's opposite to ProseMirror's philosophy, so I doubt whether we should do it.🤔Seems it's a WYSIWYG markdown editor specific feature.

**Reply by Saul-Mirone** [reply]
I think we can have a plugin for it. But I don't think we should make it as the only option.

**Reply by quank123wip** [reply]
Indeed, the editing experience should be chosen by users. Keep two well-maintained edit-styles is a good idea. And this WYSIWYG mode is friendly for users with no markdown experience, like they can just edit a document for a word-like experience in milkdown but produces markdown. But still, I prefer the mode more editors use like add identifiers like ```###``` and show them when user is focusing on them. That makes markdown editing easy to learn and doesn't require hotkeys or mouse. I think this is a important difference between a markdown editor and a common WYSIWYG software like MS Word. After all, we are editing markdown but not a HTML with markdown hotkeys and rendered by markdown. 

PS: I'm not actually making a feature request, at least not now. I think it's a complicated work not we should focus now.

**Reply by Saul-Mirone** [reply]
This is a really cool feature and I also plan to do it in the future. I think it will begin after the milkdown v6 releasing.

**Comment by quank123wip** [comment]
New issue found, when we are using inline code, for example, it's hard to insert either `` ` `` or `` \ ``. You can insert them between texts, but can't append them to the front or the back. They will escape the inline code block.

**Comment by Saul-Mirone** [comment]
> New issue found, when we are using inline code, for example, it's hard to insert either `` ` `` or `` \ ``. You can insert them between texts, but can't append them to the front or the back. They will escape the inline code block.

That's a long story. In short it's designed to do so to make sure mixing marks can be transformed to markdown correctly. Maybe play with the transformer and commonmark module can help you understand what I'm saying

**Reply by quank123wip** [reply]
Seems actually refer to the behavior markdown itself? Inserting this in github is painful, too. Anyway, I will make a guide for users avoid these problems in the future.

**Comment by quank123wip** [comment]
Seems TaskList now is clickable in docs? I think that should be read-only when milkdown is used as a renderer framwork? And seems it's a good idea to provide a API for TaskList, like providing a reference to all TaskList item in the documents, and provide a event for React/Vue/Svelte.  It's not complicated job, and I'm glad to do this in months.♥

**Reply by Saul-Mirone** [reply]
It's a bug for sure. As for the API. We don't need to create a new one because it can be calculated by data in listener, which is similar to [this](https://github.com/Saul-Mirone/vite-plugin-book/blob/main/packages/vite-plugin-book/src/hook/useEditor.ts#L66). Maybe we can provide it as an util as part of this pack: https://github.com/Saul-Mirone/milkdown/issues/387

**Reply by quank123wip** [reply]
That may not meet the need. For example, if I want to provide a list with TaskList item, and need to let other logic judge whether user can edit one of them or not, a simple listener cannot solve this. I can only come up with a hook to solve this. Did we have a better one already?

**Reply by Saul-Mirone** [reply]
Do we need let users to decide which task list item can be edited? I think we can just provide a config option for task list item to make it can be edited in read only mode.
